runtime-transaction/src/compute_budget_program_id_filter.rs

```diff
@@ -0,0 +1,27 @@
+use {solana_builtins_default_costs::MAYBE_BUILTIN_KEY, solana_sdk::pubkey::Pubkey};
+
+pub(crate) struct ComputeBudgetProgramIdFilter {
+    // array of 256 slots for all possible program_id_index (as u8),
+    // each slot indicates if a program_id_index has not been checked (eg, None),
+    // or already checked with result (eg, Some(result)) that can be reused.
+    flags: [Option<bool>; 256],
```

Comment:
See comment at: https://github.com/anza-xyz/agave/blob/master/transaction-view/src/static_account_keys_meta.rs#L13

`program_id_index` must be static account keys. Due to `PACKET_DATA_SIZE` constraint, we can only have up to 38 static account keys.

Based on that, I think we can make this array smaller.

Reply:
this assumes that the instructions we are inspecting have already been sanitized.